File: LLM Code Snippets/Gemini/JS/Task 3/Condition_A.js
const express = require('express');
const app = express();

// Assuming 'User' is your database model
const User = require('./models/User'); 

/**
 * GET /users/:id
 * Fetches a single user by their ID provided in the URL.
 */
app.get('/users/:id', async (req, res) => {
  const userId = req.params.id;

  try {
    // 1. Query the database
    const user = await User.findById(userId);

    // 2. Check if the user exists
    if (!user) {
      return res.status(404).json({
        success: false,
        message: `User with ID ${userId} not found.`
      });
    }

    // 3. Return the user record
    return res.status(200).json({
      success: true,
      data: user
    });

  } catch (error) {
    // 4. Handle unexpected errors (e.g., database connection or invalid ID format)
    console.error('Database Error:', error);
    
    return res.status(500).json({
      success: false,
      message: 'An internal server error occurred while fetching the user.'
    });
  }
});

File: LLM Code Snippets/Gemini/Python/Task 3/Condition_A.py
from flask import Flask, jsonify
from flask_sqlalchemy import SQLAlchemy

app = Flask(__name__)
app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///users.db'
db = SQLAlchemy(app)

# Define the User Model
class User(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(80), unique=True, nullable=False)
    email = db.Column(db.String(120), unique=True, nullable=False)

def get_user_by_id(user_id):
    """
    Queries the database for a user by their ID.
    Returns the user object or a JSON error response.
    """
    # .get() is the most efficient way to fetch by primary key
    user = User.query.get(user_id)
    
    if user:
        return {
            "id": user.id,
            "username": user.username,
            "email": user.email
        }, 200
    
    return {"error": "User not found"}, 404

# Example Route Usage
@app.route('/user/<int:user_id>', methods=['GET'])
def user_detail(user_id):
    data, status_code = get_user_by_id(user_id)
    return jsonify(data), status_code